.github/workflows/typescript-build-test-and-package.yml

```diff
@@ -18,8 +18,8 @@ jobs:
     env:
       NPM_TOKEN: ${{ secrets.npm-token }}
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
```

Comment:
why does changing to v3 change what node version it's running?

Reply:
Both of these actions at these major versions are ran in node16

https://github.com/actions/checkout/releases/tag/v3.0.0
https://github.com/actions/setup-node/releases/tag/v3.0.0

Reply:
wouldn't the LTS make it run on the lts version? which is 16?

i am just trying to understand what we are getting here, and what we want.  it was pointed at LTS, which should be a pointer to a moving version, which happens to be 16 right now.

so do we want to pin to run 16, or keep pointing to lts version (and when that upgrades, it will change which version we run again).

i just not quite understanding how this is stopping things from running on node12...

Reply:
So, just speaking about the `setup-node` action, yes we are still telling our job to use the LTS version of node that is available. But the `setup-node` action has scripts that they run that were running node12. You can see the change they made in the `setup-node` repo [here](https://github.com/actions/checkout/commit/25f640b3c781f672dcdc44d309d9854920dc6fa9)

Similarly, the `checkout` action by default in `v2` runs in node 12, and in runs in node 16 by default in `v3`.